### PR TITLE
020 adminapi() decorator change

### PR DESCRIPTION
**Pull request by vEpiphyte**
Allows us to suppress the logging of specific ``@adminapi()`` wrapped function.

### CODE REVIEW

File: synapse/lib/cell.py [hunk 1]
```diff
@@ -39,23 +39,29 @@
 '''
 Base classes for the synapse "cell" microservice architecture.
 '''
-def adminapi(f):
 
-    @functools.wraps(f)
-    def func(*args, **kwargs):
+def adminapi(log=True, *args, **kwargs):
```
Comment (invisig0th) [inline synapse/lib/cell.py]: I would prefer not to parameterize adminapi, but rather add a logger call to the *very* few APIs we want to actually log calls for.

Comment (jnwatson) [reply synapse/lib/cell.py]: On the fence here. It isn't *very few*. I counted 20 places that should be logged, out of 34, so more than half. It wouldn't be awful to put that in the decorator function.

Comment (vEpiphyte) [reply synapse/lib/cell.py]: per discussion - leave it as a function; remove the args/kwargs; add log=false as the the default.

File: synapse/lib/cell.py [hunk 2]
```diff
@@ -208,67 +220,67 @@ def setCellUser(self, iden):
...
         return role.pack()
 
-    @adminapi
+    @adminapi(log=True)
     async def delAuthRole(self, name):
         await self.cell.auth.delRole(name)
         await self.cell.fire('user:mod', act='delrole', name=name)
 
-    @adminapi
+    @adminapi(log=True)
```
Comment (jnwatson) [inline synapse/lib/cell.py]: no need to log this. (I bet optic will do this a bunch)

File: synapse/lib/cell.py [hunk 3]
```diff
@@ -208,67 +220,67 @@ def setCellUser(self, iden):
...
     async def getAuthUsers(self, archived=False):
         '''
         Args:
             archived (bool):  If true, list all users, else list non-archived users
         '''
         return [u.name for u in self.cell.auth.users() if archived or not u.info.get('archived')]
 
-    @adminapi
+    @adminapi(log=True)
```
Comment (jnwatson) [inline synapse/lib/cell.py]: no need to log this. (I bet optic will do this a bunch)

File: synapse/lib/cell.py [hunk 4]
```diff
@@ -208,67 +220,67 @@ def setCellUser(self, iden):
...
         await user.setAdmin(admin, gateiden=gateiden)
 
-    @adminapi
+    @adminapi(log=True)
     async def setRoleAdmin(self, name, admin, gateiden=None):
         role = await self.cell.auth.reqRoleByName(name)
         await role.setAdmin(admin, gateiden=gateiden)
 
-    @adminapi
+    @adminapi(log=True)
```
Comment (jnwatson) [inline synapse/lib/cell.py]: I think log only the mutating auth stuff.

Comment (vEpiphyte) [reply synapse/lib/cell.py]: ah yeah this was me going through all the auth stuff but the non-mutating stuff can easily be silenced.

**Pull request merged**